src/App.tsx

```diff
@@ -15,36 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetic = 'alphabetic',
+  Length = 'length',
+  Reset = 'reset',
+  Reverse = 'reverse',
+}
+
+function sortingElement(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] {
+  const sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.Alphabetic:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.Length:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+    case SortType.Reset:
+      return goodsFromServer;
+    default:
+      break;
+  }
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>();
```

Comment:
The `useState` hook for `sortField` is initialized with a type of `SortType | ''`. However, in the `handleReset` function, `setSortField` is called with an empty string, which is not a valid `SortType`. Consider initializing `sortField` with `SortType.Reset` or another valid default value.